pkg/csi/service/vanilla/controller.go

```diff
@@ -1825,15 +1829,24 @@ func (c *controller) processQueryResultsListVolumes(ctx context.Context, startin
 				}
 				// Populate List Volumes Entry Response
 				entries = append(entries, entry)
+				if volCounter == maxEntries {
+					nextTokenCounter = i + 1
+					break
+				}
 			}
 		}
 	}
 
-	// if length of queryAll entries > queryLimit, set nextToken to
-	// start_token + queryLimit
-	if len(cnsVolumes) > endIndex {
-		nextTokenInt := endIndex
-		nextToken = strconv.Itoa(nextTokenInt)
+	// for cases where number of volumes are fewer than maxEntries.
+	if nextTokenCounter == 0 {
+		// After processing all cns volumes fewer than 'maxEntries' no. of volumes were found.
+		// and i would be len(cnsVolumes).
+		// So nextTokenCounter should be set to len(cnsVolumes)
+		nextTokenCounter = len(cnsVolumes)
```

Comment:
Can we simplify this and below if block to.
if nextTokenCounter != 0 && len(cnsVolumes) > nextTokenCounter {
   // set nextToken
}

Reply:
done